extensions/resteasy-reactive/rest-common/runtime/src/main/java/io/quarkus/resteasy/reactive/common/runtime/JaxRsSecurityConfig.java

```diff
@@ -27,4 +29,15 @@ public interface JaxRsSecurityConfig {
      * annotation.
      */
     Optional<List<String>> defaultRolesAllowed();
+
+    /**
+     * Allows to run custom HTTP Security Policies after your JAX-RS resource method has been matched
+     * with incoming HTTP request. Enable this option if you have custom HTTP Security Policy where you need
+     * to inject {@link ResourceInfo} and perform authorization with the knowledge of the invoked resource method.
+     * Please review documentation of the `quarkus.http.auth.permission."permissions".applies-to`
+     * configuration property before you enable this option.
+     */
+    @WithName("enable-jaxrs-http-security-policies")
```

Comment:
Hmm. But `applies-to=JAX-RS` has been about delaying the policy before the JAX-RS has been matched.
And I'm not sure we need to have a ResourceInfo dedicated support.

Reply:
I thought that user scenario is legit, even though very much edge case. Not sure what else can be said to this @sberyozkin .

Reply:
I mean, it's not like it will cost us anything, it's one boolean. The change inside `EagerSecurityHandler` is something that needs to be fixed anyway, and so is inside HTTP Authorizer. I am fixing it as part of this, because here, I have tests that discovered it. I'll create simple dedicated backport for the fixes.

Reply:
> But applies-to=JAX-RS has been about delaying applying the policy until after the JAX-RS resource has been matched.

can you elaborate, please? it still is. whole point of this PR is that `applies-to=JAXRS` can only be used when we know during the build time that it will be used.

Reply:
>I thought that user scenario is legit, even though very much edge case

I agree it is a legit, it just felt it was not a common enough to specifically highlight in the JavaDocs. May be it should be highlighted, I don't now

>whole point of this PR is that applies-to=JAXRS can only be used when we know during the build time that it will be used inside Quarkus REST. We can detect @Tenant annotation, so we know that applies-to=JAXRS may be needed, but we can't detect runtime property applies-to=JAXRS during the build time, hence this PR is adding a flag.

Maybe I'm missing something but I feel like we keeping adding one property upon another with
```
quarkus.http.auth.permission.custom-policy-1.applies-to=JAXRS
quarkus.security.jaxrs.enable-jaxrs-http-security-policies=true
```

The user says `quarkus.http.auth.permission.custom-policy-1.applies-to=JAXRS`, therefore this policy must not run before the JAX-RS is matched. IMHO we don't need to worry about finding some build-time confirmation that it will be necessary, the user said it already it is necessary. May be JAX-RS won't have `@Tenant` or something else, this policy will still work.

Reply:
>Maybe I'm missing something but I feel like we are keeping adding one property upon another with

I dare to say this is just misunderstanding. See below.

> The user says `quarkus.http.auth.permission.custom-policy-1.applies-to=JAXRS`, therefore this policy must not run before the JAX-RS is matched. IMHO we don't need to worry about finding some build-time confirmation that it will be necessary, the user said it already it is necessary.

User said it during the **_runtime_**, because `quarkus.http.auth.permission.custom-policy-1.applies-to=JAXRS`. But we need to know it during the build time. When there is at least one instance of the `@Tenant`, then we know and we enable it.

@sberyozkin this buildtime /runtime thingy is difference between adding serverresthandler to every single endpoint always or only on demand. I think adding one build-time flag for legit edge case is acceptable.

> May be JAX-RS won't have `@Tenant` or something else, this policy will still work.

It can't work without knowing at the build time. Hence, the flag.

Reply:
I am open to any suggestions like drop even more of documentation (the javadoc) etc. but this solution is not me being lazy finding a good solution :-) there simply is none, we either add that handler always (and I strongly believe that would be a waste to do that for edge case), or we don't.

Reply:
@michalvavrik 
>but this solution is not me being lazy finding a good solution :-)

Michal, without you Quarkus Security would be quite a few steps behind, I'm not trying to suggest anything like that. Let me ping you offline to clarify a couple of details